storage/google/cloud/storage/bucket.py

```diff
@@ -272,6 +272,82 @@ def from_api_repr(cls, resource):
         return instance
 
 
+class IAMConfiguration(dict):
```

Comment:
I agree that a single name is more consistent with bucket lock:  I was looking at the lifecycle policy bits, but they aren't as close a fit.  I will update.

Reply:
@frankyn I likely had that in mind with the original implementation.  If you already know *that* such changes will happen (if not when), then we should revert 6814a4f.  Otherwise, I think we can call YAGNI, leaving the simpler surface, and move to the "helper object" version in the future, when we would deprecate `iam_configuration_bucket_policy_only` and `iam_configuration_locked_time` in favor of `iam_configuration`.

Reply:
I'd prefer not to deprecate methods in the future. There's an expectation to increase the number of field metadata for iamConfiguration in the future per design discussions. Please revert to original design. Thank you for your patience.

Reply:
3b43092